R/Rules-validity.R

```diff
@@ -16,19 +16,43 @@ NULL
 validate_increments_numdoselevels <- function(object) {
   o <- Validate()
   o$check(
-    is.scalar(object@maxLevels) && 
-    is.integer(object@maxLevels) && 
+    is.scalar(object@maxLevels) &&
+    is.integer(object@maxLevels) &&
     object@maxLevels > 0,
     "maxLevels must be scalar positive integer"
   )
   o$check(
-    is.scalar(object@basisLevel) && 
+    is.scalar(object@basisLevel) &&
     object@basisLevel %in% c("last", "max"),
     "basisLevel must be either 'last' or 'max'"
   )
   o$result()
 }
 
+#' @describeIn validate_stopping validates that the [`IncrementsHSRBeta`]
+#'  object contains valid probability target, threshold and shape parameters.
+validate_increments_hsr_beta <- function(object) {
+  o <- Validate()
+  o$check(
+    is.scalar(object@target) & is.probability(object@target, bounds = FALSE),
+    "target must be a probability skalar"
```

Comment:
```suggestion
    is.probability(object@target, bounds = FALSE),
    "target must be a probability"
```

Reply:
not needed here, see https://github.com/Roche/crmPack/blob/022f5f62d194e81f8297058d90a61e61309c206e/R/helpers.R#L158 already checks for scalar

Reply:
> last comments :-)

Don't worry. Your comments are much appreciated. They all make much sense and helps me to understand what to do in the future.